src/components/List.js

```diff
@@ -105,6 +121,7 @@ const List = ({ token }) => {
       )}
 
       {errorMessage && <Error errorMessage={errorMessage} />}
+      {deleteError && <Error errorMessage={deleteError} />}
```

Comment:
Will there ever be multiple different errors at once? It might be more convenient to just reuse the existing error message functionality instead of repeating yourself for each kind of error type.

Reply:
Good question. I think right now the `errorMessage` on line 123 is coming from the `useFirestore` hook, so we weren't sure how to integrate any errors coming from our delete action to make the code re-usable. I feel like I'm missing something about how we could be using `useFirestore`...

Reply:
One thing you could do is have useFirestore actually do full CRUD operations. You could add a `deleteDoc` function that looks something like 
```javascript
const deleteDoc = item =>
 try {
        db.collection(token).doc(item.id).delete();
      } catch (error) {
        setErrorMessage(`Problem deleting ${item.itemName}`);
     }
```
and return that from useFirestore. Then, inside of `List`, you can call `deleteDoc` inside of `confirmDelete`. You will need some way to clear the error as was mentioned elsewhere. You could return `setError` from `useFireStore` and use that to clear the error. Or you could do something fun like have the error just clear itself in a setTimeout so that it looks something like 
```javascript
  const deleteDoc = (doc) => {
    try {
      db.collection(collection).doc(doc.id).delete();
    } catch {
      setErrorMessage(`Problem deleting ${item.itemName}`);
      setTimeout(() => {
        setErrorMessage('');
      }, 3000);
    }
  };
```
Or you could play around with something like https://www.npmjs.com/package/react-alert. All sorts of fun options!

Reply:
I played around with this and was able to get everything working, moving the delete functionality into the `useFirestore` hook and just calling the `deleteDoc` function inside the `confirmDelete` function. That way we are able to reuse the Error component, without setting new state in the case of a delete error.
Brittany and I discussed how the error message should be cleared, would it be better for the user to clear it themselves or for us to clear it with something like the setTimeout function you mentioned?